src/lib/optimizer/join_ordering/enumerate_ccp.cpp

```diff
@@ -36,44 +38,42 @@ EnumerateCcp::EnumerateCcp(const size_t num_vertices, std::vector<std::pair<size
 
 std::vector<std::pair<JoinGraphVertexSet, JoinGraphVertexSet>> EnumerateCcp::operator()() {
   /**
-   * Initialize vertex neighborhood lookup table by computing and storing the neighborhood of each vertex
+   * Initialize vertex neighborhood lookup table by computing and storing the neighborhood of each vertex.
    */
   _vertex_neighborhoods.resize(_num_vertices);
   for (auto vertex_idx = size_t{0}; vertex_idx < _num_vertices; ++vertex_idx) {
     _vertex_neighborhoods[vertex_idx] = _single_vertex_neighborhood(vertex_idx);
   }
 
   /**
-   * This loop corresponds to EnumerateCsg in the paper
+   * This loop corresponds to EnumerateCsg in the paper.
    *
-   * It iterate from the highest to the lowest vertex index and starts a search for connected subgraphs from
-   * each vertex (_enumerate_csg_recursive()).
-   * For each subgraph, a search for complement subgraphs is started (_enumerate_cmp()).
+   * It iterates from the highest to the lowest vertex index and starts a search for connected subgraphs from
+   * each vertex (`_enumerate_csg_recursive()`). For each subgraph, a search for complement subgraphs is started
+   * (`_enumerate_cmp()`).
    */
-  for (auto reverse_vertex_idx = size_t{0}; reverse_vertex_idx < _num_vertices; ++reverse_vertex_idx) {
-    const auto forward_vertex_idx = _num_vertices - reverse_vertex_idx - 1;
-
+  for (auto vertex_idx = _num_vertices; vertex_idx > 0; --vertex_idx) {
     auto start_vertex_set = JoinGraphVertexSet(_num_vertices);
```

Comment:
Can we pull `auto start_vertex_set` out of the loop (it's always the same size) and just resize in each row? Saves some allocations.

Reply:
Okay, its `reset` rather than `clear`. Gonna measure if that's faster than allocating.

Reply:
Did the following:
https://github.com/hyrise/hyrise/blob/bd6c2e649fa397ecae38370f97619fbe35c2650d/src/bin/playground.cpp#L15-L57

The second version was faster on Mac and nemea. Thus, let's use a single bitset with `reset`.